packages/manager/apps/pci-kubernetes/src/pages/detail/nodepools/new/store.ts

```diff
@@ -0,0 +1,214 @@
+  isMonthlyBilling: boolean;
+  steps: Map<StepsEnum, TStep>;
+  set: {
+    name: (val: string) => void;
+    flavor: (val: TFlavor) => void;
+    autoScaling: (val: AutoscalingState) => void;
+    antiAffinity: (val: boolean) => void;
+    isMonthlyBilling: (val: boolean) => void;
+  };
+  open: (step: StepsEnum) => void;
+  close: (step: StepsEnum) => void;
+
+  lock: (step: StepsEnum) => void;
+  unlock: (step: StepsEnum) => void;
+
+  check: (step: StepsEnum) => void;
+  uncheck: (step: StepsEnum) => void;
+
+  edit: (step: StepsEnum) => void;
+
+  reset: () => void;
+};
+
+const initialSteps = new Map<StepsEnum, TStep>([
+  [
+    StepsEnum.NAME,
+    {
+      isOpen: true,
+      isLocked: false,
+      isChecked: false,
+    },
+  ],
+  [
+    StepsEnum.TYPE,
+    {
+      isOpen: false,
+      isLocked: false,
+      isChecked: false,
+    },
+  ],
+  [
+    StepsEnum.SIZE,
+    {
+      isOpen: false,
+      isLocked: false,
+      isChecked: false,
+    },
+  ],
+  [
+    StepsEnum.BILLING,
+    {
+      isOpen: false,
+      isLocked: false,
+      isChecked: false,
+    },
+  ],
+]);
+
+export const useNewPoolStore = create<TFormStore>()((set, get) => ({
+  name: '',
+  flavor: undefined,
+  autoScaling: null,
+  antiAffinity: false,
+  isMonthlyBilling: false,
+  steps: new Map(initialSteps),
+  set: {
+    name: (val: string) => set({ name: val }),
+    flavor: (val: TFlavor) => set({ flavor: val }),
+    autoScaling: (val: AutoscalingState) => set({ autoScaling: val }),
+    antiAffinity: (val: boolean) => set({ antiAffinity: val }),
+    isMonthlyBilling: (val: boolean) => set({ isMonthlyBilling: val }),
+  },
+  open: (id: StepsEnum) => {
+    set((state) => {
+      const steps = new Map(state.steps);
+      steps.get(id).isOpen = true;
+      return {
+        ...state,
+        steps,
+      };
+    });
+  },
+  close: (id: StepsEnum) => {
+    set((state) => {
+      const steps = new Map(state.steps);
+      steps.get(id).isOpen = false;
+      return {
+        ...state,
+        steps,
+      };
+    });
+  },
+  lock: (id: StepsEnum) => {
+    set((state) => {
+      const steps = new Map(state.steps);
+      steps.get(id).isLocked = true;
+      return {
+        ...state,
+        steps,
+      };
+    });
+  },
+  unlock: (id: StepsEnum) => {
+    set((state) => {
+      const steps = new Map(state.steps);
+      steps.get(id).isLocked = false;
+      return {
+        ...state,
+        steps,
+      };
+    });
+  },
+  check: (id: StepsEnum) => {
+    set((state) => {
+      const steps = new Map(state.steps);
+      steps.get(id).isChecked = true;
+      return {
+        ...state,
+        steps,
+      };
+    });
+  },
+  uncheck: (id: StepsEnum) => {
+    set((state) => {
+      const steps = new Map(state.steps);
+      steps.get(id).isChecked = false;
+      return {
+        ...state,
+        steps,
+      };
+    });
+  },
+  edit: (id: StepsEnum) => {
+    // Activate
+    get().open(id);
```

Comment:
nitpick (optional) : i think it's more natural to declare a variable instead of calling "get()" repeatidly
```
const store = get();

store.uncheck(...);
store.unlock(...):
```

Reply:
I believe that since it's a function, the value may change between calls, there must be a reason behind the fact that it's a function instead of simple variable